internal/gke/update.go

```diff
@@ -33,31 +33,32 @@ const (
 // UpdateMasterKubernetesVersion updates the Kubernetes version for the control plane.
 // This must occur before the Kubernetes version is changed on the nodes.
 func UpdateMasterKubernetesVersion(ctx context.Context, client *gkeapi.Service, config *gkev1.GKEClusterConfig, upstreamSpec *gkev1.GKEClusterConfigSpec) (Status, error) {
-	if config.Spec.KubernetesVersion == nil {
+	if config.Spec.KubernetesVersion == nil || *config.Spec.KubernetesVersion == "" {
```

Comment:
I think it would be nice to use the `utils` function since we have it.
```suggestion
	if utils.StringValue(config.Spec.KubernetesVersion) == "" {
```

Reply:
I should have said for all of these.

Reply:
done